apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -278,8 +278,25 @@ type BareMetalHostSpec struct {
 	// the power status and hardware inventory inspection. If the
 	// Image field is filled in, this field is ignored.
 	ExternallyProvisioned bool `json:"externallyProvisioned,omitempty"`
+
+	// When set to Disabled, automated cleaning will be avoided
+	// during provisioning and deprovisioning.
+	// +optional
+	// +kubebuilder:default:=Full
+	// +kubebuilder:validation:Optional
+	AutomatedCleaningMode AutomatedCleaningMode `json:"automatedCleaningMode,omitempty"`
 }
 
+// AutomatedCleaningMode is the interface to enable/disable automated cleaning
+// +kubebuilder:validation:Enum:=Full;Disabled
+type AutomatedCleaningMode string
+
+// Allowed automated cleaning modes
+const (
+	Disabled AutomatedCleaningMode = "Disabled"
```

Comment:
It would be good to namespace these constants a bit like we do with the state machine states. I know we didn't do that with the checksum types just below, but we probably should have.

```suggestion
	CleaningModeDisabled AutomatedCleaningMode = "Disabled"
```

Reply:
Good point, fixed it.